### ISSUE TITLE
matchSignatures color might cause misunderstanding

### ISSUE THREAD

**Issue by jmche** [dami82/mutSignatures#8]
I saw the raw code of matchSignatures and I think the function's color might cause misunderstanding.
1. The distance is cosine distance and It's more different if the cosine distance is close to 1 and more similar if the distance is close to 0
2. The color is red if it's close to 1.
3. People would think the red shows more similar, however from the code It is not.
I think it's better to show the similarity in the plot instead of the distance

**Comment by dami82** [comment]
Thanks for the comment. Your point is well taken, I guess that using **similarity** (defined as `1 - distance`) instead of distance in the heatmap makes sense. I agree that it is more intuitive to look for signal instead of *absence of signal*. I will work on the issue you opened in the next few weeks (sorry, but I really have no bandwidth in the next ~3 weeks).

However, it is **inaccurate** to say that in its current implementation, my code returns *red* when the distance is ~1. You may have missed the following line:
```
p  <- p + ggplot2::scale_fill_gradient2(low = "#f40000", mid = "#ffe9e9", high = "white",
                                         midpoint = nuMid, guide = "colourbar",
                                         name = paste(method, "\ndistance", sep = "")
```
Briefly, this flips the color palette so that the color is **white (high) when the cosine distance is ~1** AND the color is **red (#f40000) when the distance is low or close to zero (*i.e.*, high similarity)**.
You may want to verify this by matching few COSMIC signatures against themselves. Red boxes should highlight the diagonal.

**Note**: *this may seem like a trivial modification of the code, but it may take a little more work than you anticipate cause the `cosine distance` is the default distance method, but other methods are also supported. Therefore, when I modify the function, I need to make sure that the similarity scores [max.distance - distance_i] are computed accurately, and that the final implementation is reliable independently of the desired distance method.*

**Comment by jmche** [comment]
Yeah, I think I was misunderstanding the color and the value of cosine distance. I didn't see the legend details that's why I thought the red means larger cosine distance and it made me really confused. 
According to what you said and I think the plot is OK, however, using similarity will be better to avoid the misunderstanding just like I did.
